Create deploying-django-application-non-docker.md (#680)
porter-dev/porter: docs/deploy/applications/deploying-django-application-non-docker.md

# Deploy Django Application (Non-Dockerize)

To deploy your Django Application in Porter. You need to tweak something on your Django Application.

> 📘 Prerequisites
> - Django Application

> > - Docker Registry Integrate on your account ( Work only on ACR, GCR, DOCR )

Reviewer: ```suggestion
> - Docker Registry integration on your account ([See docs](https://docs.getporter.dev/docs/linking-an-existing-docker-container-registry))
```

## Prepare Django Application

> 1. Install django-allow-cidr ( A Django Middleware to enable use of CIDR IP ranges in ALLOWED_HOSTS )

Reviewer: ```suggestion
1. Install `django-allow-cidr` (Django middleware to enable the use of CIDR IP ranges in `ALLOWED_HOSTS`)
```

  ```sh
  pip install django-allow-cidr
  ```
2.  Go to Django Settings and add os.environ.get in allowed host.
  ```python
  ALLOWED_HOSTS = os.environ.get("DJANGO_ALLOWED_HOSTS", default='127.0.0.1').split(" ")
  ```
3. Add this below allowed host. Put CIDR accordning to the K8s kubelet CIDR
  ```python

>   ALLOWED_CIDR_NETS = os.environ.get("ALLOWED_CIDR_NETS", default='10.0.0.0/8').split(" ")

Reviewer: ```suggestion
  ALLOWED_CIDR_NETS = os.environ.get("ALLOWED_CIDR_NETS", default='10.0.0.0/16').split(" ")
```

The k8s CIDR is usually `10.0.0.0/16`, so this might be a better default?

  ```

> 4. Add django-allow-cidr middleware on the top of django middleware

Reviewer: ```suggestion
4. Add `django-allow-cidr` middleware on the top of Django middleware:
```

  ```python
  MIDDLEWARE = [
    'allow_cidr.middleware.AllowCIDRMiddleware',
    #'django.middleware.security.SecurityMiddleware',
  ]
  ```
5. Add Gunicorn
  ```sh
  pip install gunicorn
  ```

> 6. Add static folder then put your html and css files. Locate static url settings and add static file dirs below

Reviewer: ```suggestion
6. Add static folder and add your HTML and CSS files. Locate static URL settings and add static file dirs below:
```

  ```python
  STATICFILES_DIRS = (
    os.path.join(BASE_DIR, 'static'),
  )
  STATIC_ROOT = os.path.join(BASE_DIR, 'staticfiles')
  ```

> 7. Add Procfile file and add this

Reviewer: ```suggestion
7. Add Procfile and add this:
```

  ```

>   web: gunicorn <project-name>.wsgi -b 0.0.0.0:8989 --timeout 120 (ex. web: gunicorn djangosample.wsgi -b 0.0.0.0:8989 --timeout 120 )

Reviewer: ```suggestion
web: gunicorn <project-name>.wsgi -b 0.0.0.0:8989 --timeout 120 
  
For example: 
  
web: gunicorn djangosample.wsgi -b 0.0.0.0:8989 --timeout 120
```
